File: 2017/01/part1.js
const fs = require("fs");

fs.readFile("input.txt", "utf8", (err, input) => {
  let sum = 0;
  const digits = input.split("").map(Number);

  for (let i = 0; i < digits.length; i++) {
    if (digits[i] === digits[(i + 1) % digits.length]) {
      sum += digits[i];
    }
  }

  console.log(sum);
});

File: 2023/02/part1.py
bags = {
    "red": 12,
    "green": 13,
    "blue": 14,
}

with open("input.txt", "r") as file:
    count = 0

    for line in file:
        line = line.strip()
        [number, rounds] = line.split(': ')
        flag = False
        for round in rounds.split(';'):
            for draw in round.split(','):
                [quantity, color] = draw.strip().split(' ')
                if bags[color] < int(quantity):
                    flag = True
                    break
        if flag == False:
            count += int(number.split(' ')[1])


print(count)
